cmd/influx_tools/parquet/exporter.go

```diff
@@ -0,0 +1,488 @@
+			}
+
+			switch strings.ToLower(ftype) {
+			case "float":
+				e.typeResolutions[measurement][field] = influxql.Float
+			case "int":
+				e.typeResolutions[measurement][field] = influxql.Integer
+			case "uint":
+				e.typeResolutions[measurement][field] = influxql.Unsigned
+			case "bool":
+				e.typeResolutions[measurement][field] = influxql.Boolean
+			case "string":
+				e.typeResolutions[measurement][field] = influxql.String
+			default:
+				return nil, fmt.Errorf("invalid type in conflict resolution %q", r)
+			}
+		}
+	}
+
+	// Prepare name resolutions
+	if cfg.NameResolutions != "" {
+		for _, r := range strings.Split(cfg.NameResolutions, ",") {
+			field, name, found := strings.Cut(r, "=")
+			if !found {
+				return nil, fmt.Errorf("invalid format in name conflict resolution %q", r)
+			}
+			measurement, field, found := strings.Cut(field, ".")
+			if !found {
+				return nil, fmt.Errorf("invalid measurement in name conflict resolution %q", r)
+			}
+			if _, exists := e.nameResolutions[measurement]; !exists {
+				e.nameResolutions[measurement] = make(map[string]string)
+			}
+			e.nameResolutions[measurement][field] = name
+		}
+	}
+	return e, nil
+}
+
+func (e *exporter) open(ctx context.Context) error {
+	if err := e.store.Open(); err != nil {
+		return err
+	}
+
+	// Determine all shard groups in the database
+	min := time.Unix(0, models.MinNanoTime)
+	max := time.Unix(0, models.MaxNanoTime)
+	groups, err := e.client.NodeShardGroupsByTimeRange(e.db, e.rp, min, max)
+	if err != nil {
+		return err
+	}
+
+	if len(groups) == 0 {
+		return nil
+	}
+
+	sort.Sort(meta.ShardGroupInfos(groups))
+	e.startDate = groups[0].StartTime
+	e.endDate = groups[len(groups)-1].EndTime
+	e.groups = groups
+
+	// Collect all shards
+	for _, grp := range groups {
+		ids := make([]uint64, 0, len(grp.Shards))
+		for _, s := range grp.Shards {
+			ids = append(ids, s.ID)
+		}
+		e.shards = append(e.shards, e.store.Shards(ids)...)
+	}
+
+	// Determine all measurements in all shards
+	if len(e.measurements) == 0 {
+		measurements := make(map[string]bool)
+		for _, shard := range e.shards {
+			if err := shard.ForEachMeasurementName(func(name []byte) error {
+				measurements[string(name)] = true
+				return nil
+			}); err != nil {
+				return fmt.Errorf("getting measurement names failed: %w", err)
+			}
+		}
+		for m := range measurements {
+			e.measurements = append(e.measurements, m)
+		}
+	}
+	sort.Strings(e.measurements)
+
+	// Collect the schemata for all measurments
+	e.schemata = make(map[string]*schemaCreator, len(e.measurements))
+	for _, m := range e.measurements {
+		creator := newSchemaCreator(m, e.shards, e.typeResolutions[m], e.nameResolutions[m])
+		if err := creator.extractSchema(ctx); err != nil {
+			return fmt.Errorf("extracting schema for measurement %q failed: %w", m, err)
+		}
+		e.schemata[m] = creator
+	}
+
+	return nil
+}
+
+func (e *exporter) close() error {
+	return e.store.Close()
+}
+
+func (e *exporter) printPlan() {
```

Comment:
This can fail and maybe should return an error

Reply:
How can this fail? If you are taking about the error when calling `creator.validate()`, then this is expected as this will just signal that the schema is not valid and has unresolved conflicts. The actual error handling is then done in `export()`...